src/wirecloud/commons/search.py

```diff
@@ -0,0 +1,365 @@
+GROUPS_INDEX = 'groups'
+
+USER_CONTENT_FIELDS = ['fullname', 'username']
+GROUP_CONTENT_FIELDS = ['name']
+
+USER_MAPPINGS = {
+    "settings": {
+        "analysis": {
+            "analyzer": {
+                "ngram_analyzer": {
+                    "tokenizer": "ngram_tokenizer",
+                    "filter": ["lowercase"]
+                },
+                "keyword_lowercase": {
+                    "tokenizer": "keyword",
+                    "filter": ["lowercase"]
+                }
+            },
+            "tokenizer": {
+                "ngram_tokenizer": {
+                    "type": "ngram",
+                    "min_gram": 2,
+                    "max_gram": 20,
+                    "token_chars": ["letter", "digit"]
+                }
+            }
+        }
+    },
+    "mappings": {
+        "properties": {
+            "fullname": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard",
+                "fields": {
+                    "keyword": {"type": "keyword"}
+                }
+            },
+            "username": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard",
+                "fields": {
+                    "keyword": {"type": "keyword"}
+                }
+            }
+        }
+    }
+}
+GROUP_MAPPINGS = {
+    "settings": {
+        "analysis": {
+            "analyzer": {
+                "ngram_analyzer": {
+                    "tokenizer": "ngram_tokenizer",
+                    "filter": ["lowercase"]
+                },
+                "keyword_lowercase": {
+                    "tokenizer": "keyword",
+                    "filter": ["lowercase"]
+                }
+            },
+            "tokenizer": {
+                "ngram_tokenizer": {
+                    "type": "ngram",
+                    "min_gram": 2,
+                    "max_gram": 20,
+                    "token_chars": ["letter", "digit"]
+                }
+            }
+        }
+    },
+    "mappings": {
+        "properties": {
+            "name": {
+                "type": "text",
+                "analyzer": "ngram_analyzer",
+                "search_analyzer": "standard",
+                "fields": {
+                    "keyword": {"type": "keyword"}
+                }
+            }
+        }
+    }
+}
+
+_available_search_engines = None
+_available_rebuild_engines = None
+
+
+class UserOut(BaseModel):
+    fullname: str
+    username: str
+    # TODO: Organizations
+
+
+class GroupOut(BaseModel):
+    name: str
+    # TODO: Organizations
+
+
+class SearchResponse(BaseModel):
+    offset: int
+    pagecount: int
+    pagelen: int
+    pagenum: int
+    results: list[Union[UserOut, GroupOut, WorkspaceOutResponse, ResourceOutResponse]]
+    total: int
+
+
+def get_available_search_engines() -> dict[str, Callable[[str, int, int, str | None], SearchResponse]]:
```

Comment:
Cuidado! La sintaxis `str | None` es de python 9 y tratamos de soportar python 8. Cámbialo por `Union[str, None]`, o aun mejor `Optional[str]`.

```suggestion
def get_available_search_engines() -> dict[str, Callable[[str, int, int, Optional[str]], SearchResponse]]:
```